packages/svelte/src/internal/client/runtime.js

```diff
@@ -797,7 +797,16 @@ export function invalidate_inner_signals(fn) {
 		captured_signals = previous_captured_signals;
 	}
 	for (signal of captured) {
-		mutate(signal, null /* doesnt matter */);
+		// Go one level up because derived signals created as part of props in legacy mode
```

Comment:
this is a clever trick though i wonder if we should restrict it to the new deriveds — maybe we add a new flag? feels like we could invalidate too broadly otherwise

Reply:
True